numbast/src/numbast/struct.py

```diff
@@ -148,6 +148,20 @@ def ctor_impl(context, builder, sig, args):
         )
         return builder.load(selfptr, align=getattr(s_type, "alignof_", None))
 
+    if ctor.kind == method_kind.converting_constructor:
+
+        @lower_cast(*param_types, s_type)
```

Comment:
Are we certain `param_types` is length 1 here?

Reply:
Should we add a line `assert(len(param_types) == 1, "isConvertinConstructor in clang ensures that only one parameter is passed"` here to avoid surprises?